**Fix**: Model Confirm Styling

ericadamski opened the pull request:
<!-- 
  ❗️IMPORTANT ❗️
  Please prefix the title of this PR with _one_ of the following.

  **Breaking:**
    For when we break a public API.

  **Feature:** 
    For when we add something NEW that doesn't
    break the public API.
      
  **Fix:**
    For when we fix something that previously
    did not look or work correctly.
    
  Leaving off this prefix will prevent the PR from being
  included in a release. A good case to omit the prefix
  is when proposing infrastructural changes to the repo
  that do not affect the library.
-->
# Summary

As mentioned in #698 the modal styling had been previously broken. This PR serves to patch it. Below are some screenshots of the patched renders of the model on `styleguidist` and `dev` respectfully.

![screen shot 2018-09-05 at 8 05 54 am](https://user-images.githubusercontent.com/6516758/45092210-baed2000-b0e2-11e8-83c9-fd80a5f362ae.png)
![screenshot_2018-09-05 screenshot](https://user-images.githubusercontent.com/6516758/45092215-be80a700-b0e2-11e8-9f34-a7e0454dddd0.png)

# Related issue

#698 

# To be tested

Me
- [x] No error or warning in the console on `localhost:6060`

Tester 1

- [x] Things look good on the demo.
  <!-- Put here everything that the reviewer 1 should test to be sure that everything is working properly -->

Tester 2

- [ ] Things look good on the demo.
  <!-- Put here everything that the reviewer 2 should test to be sure that everything is working properly -->

ericadamski commented:
@TejasQ this is what happens on CI 😢 
![screen shot 2018-09-05 at 8 27 13 am](https://user-images.githubusercontent.com/6516758/45093156-90e92d00-b0e5-11e8-8b2b-6455d068a012.png)

TejasQ commented:
Thanks, Eric. It's strange though because it's _in_ the build config.

![image](https://user-images.githubusercontent.com/9947422/45093299-47fe9c00-b118-11e8-8132-e9fd5f67a448.png)

ericadamski commented:
That is odd. 🤔

TejasQ commented:
@ericadamski alright, I think I've fixed it on master. Mind rebasing on more time for me?

ericadamski commented:
🎉 🎉 🎉  Nice fix @TejasQ !
![screenshot_2018-09-05 fix model confirm styling by ericadamski pull request 723 contiamo operational-ui](https://user-images.githubusercontent.com/6516758/45094063-3c937c80-b0e8-11e8-9396-c31085cd55bf.png)

peterszerzo commented:
@ericadamski looking great! The absolute positioning will cause the actions to cover the text like in the screenshot below. Now since this is a confirm component with ideally not much text, it's certainly not worth breaking our neck to support this, but just in case anyone abuses the space, I think it would be safer if the actions stayed in the document flow and would only be reachable by scrolling down in case the content is too tall.

<img width="722" alt="screen shot 2018-09-05 at 3 01 04 pm" src="https://user-images.githubusercontent.com/6738398/45094784-99a92580-b11c-11e8-8ada-bedf181b8e4e.png">

fabien0102 commented:
@peterszerzo and me are so sync! I'm afraid ^^

ericadamski commented:
I am not super thrilled with this fix. I don't have a great grasp of the organization and hierarchy of the library so making this change to the `Card` generic component feels a little odd. Here are some screenshots of the results and one side effect of the change to `Card`

Fixed with short message:
<img width="651" alt="screen shot 2018-09-05 at 9 34 35 pm" src="https://user-images.githubusercontent.com/6516758/45130225-d2192580-b155-11e8-8e6e-266bbd7a014a.png">
Fixed with longer message:
<img width="662" alt="screen shot 2018-09-05 at 9 34 43 pm" src="https://user-images.githubusercontent.com/6516758/45130226-d2192580-b155-11e8-9996-df88abbcfe6d.png">

Original before fix:
<img width="629" alt="screen shot 2018-09-05 at 9 44 11 pm" src="https://user-images.githubusercontent.com/6516758/45130228-d2b1bc00-b155-11e8-999e-8cd936098a6b.png">

Issue after fix:
<img width="1113" alt="screen shot 2018-09-05 at 9 44 26 pm" src="https://user-images.githubusercontent.com/6516758/45130229-d2b1bc00-b155-11e8-8f28-53336d2c3c9c.png">

Possible solution?
<img width="1097" alt="screen shot 2018-09-05 at 9 44 41 pm" src="https://user-images.githubusercontent.com/6516758/45130230-d2b1bc00-b155-11e8-9d65-fbbb796d2b1a.png">

ericadamski commented:
:100: I prefer that for sure!

ericadamski commented:
@TejasQ how important is the `min-height` on the `ControlledModal`?

https://github.com/contiamo/operational-ui/blob/ceacae831399c56ef94912f08700a1673ddd2559/src/Internals/ControlledModal.tsx#L34

ericadamski commented:
This is without the `min-height`
![screen shot 2018-09-06 at 6 58 21 am](https://user-images.githubusercontent.com/6516758/45153518-4aa5d380-b1a2-11e8-981a-b32b3f7f909e.png)

fabien0102 commented:
To add more context, the `min-height` reflect our design specs.

The screenshot for our internal zeplin:
![image](https://user-images.githubusercontent.com/1761469/45153621-efd99f80-b1d4-11e8-9199-9d0c71d6c906.png)

TejasQ commented on src/Internals/Confirm.tsx:
```diff
@@ -1,10 +1,10 @@
 import ControlledModal from "./ControlledModal"
 
 const Actions = styled("div")({
+  marginTop: 20,
```
Actually, there's a `theme` variable here.

If this object literal is instead a function, it receives props as its first argument. Then, you can use `props.theme.space.element` here in order to be really adherent to our designs. 

Great work, Eric!
